### PR TITLE
WIP: Support for JRuby-9000

### PR DESCRIPTION
**Pull request by kennethkalmer**
Will bring in the combined efforts of previous attempts to get activeadmin running on jruby-9000, slated to be the next major JRuby release.

- [x] Get dependencies installed
- [x] Get the tests running
- [ ] Fix any test failures
- [x] Hook up to CI
- [ ] Prosper

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -22,16 +22,17 @@ gem 'pry'                  # Easily debug from your console with `binding.pry`
...
-  gem 'flamegraph'         # Flamegraph visualiztion: ?pp=flamegraph
+  gem 'rack-mini-profiler'          # Inline app profiler. See ?pp=help for options.
+  gem 'flamegraph', platforms: :mri # Flamegraph visualiztion: ?pp=flamegraph
 
   # Documentation
-  gem 'yard'               # Documentation generator
-  gem 'redcarpet'          # Markdown implementation (for yard)
+  gem 'yard'                        # Documentation generator
+  gem 'redcarpet', platforms: :mri  # Markdown implementation (for yard)
+  gem 'kramdown', platforms: :jruby # Markdown implementation (for yard)
```
Comment (timoschilling) [inline Gemfile]: please bring all comments of all blocks into one line

Comment (kennethkalmer) [inline Gemfile]: across the entire Gemfile? Just making sure before I do

Comment (timoschilling) [inline Gemfile]: I think across the entire Gemfile is a good idea

**Comment by brennovich** [comment]
I'll try to help with this issue :)

Although I have a concern: will **Rails** `3.2.X` be supported? If so, I don't think we'll be able to accomplish that.

Besides that I'll start with **Rails** `4.2.0`.

**Comment by brennovich** [comment]
Well, well all *unit* spec seems to be passing, I needed to increase heap size though

screenshot 2015-02-15 19 47 44

**Comment by kennethkalmer** [comment]
@brennovich thanks! I bumped the heap to 1024 and it all passed, but on Travis it timed out after 50 mins :(

I guess we need to find the source of Test::Unit's AutoRunner kicking in next.

There are also some specs using `ObjectSpace` which I expected to fail off the bat, but either they work or they simply haven't run yet.

**Comment by timoschilling** [comment]
Is this done? Travis shows it up as green.

**Comment by timoschilling** [comment]
can you please move jruby from the `allow_failures` list to the normal list. So that this PR is listet as failing in GH until you are done.

**Comment by kennethkalmer** [comment]
@timoschilling not yet, I've been soliciting feedback to help see this through. That said, let me put in some time today to merge in the mainline branches and run tests again. Thanks for the push to get his merged!

**Comment by timoschilling** [comment]
It's ok that you still need time for this PR. I only want to see the true CI status in the PR List like it is on #3742 
pull_requests_ _activeadmin_activeadmin

**Comment by deivid-rodriguez** [comment]
I tried `activeadmin` with JRuby today. It's working.

I rebased this PR and got only a couple of test failures:

```
Failing Scenarios:
cucumber features/index/batch_actions.feature:3 # Scenario: Use default (destroy) batch action
cucumber features/index/batch_actions.feature:40 # Scenario: Use default (destroy) batch action on a nested resource
```

The functionality they test actually works, so it seems like a bug / misuse of capybara / poltergeist.

**Comment by deivid-rodriguez** [comment]
If I add the `@javascript` tag on top of those scenarios, those assertions pass, but I get other failures because the assertions are not supported by the poltergeist driver.

**Comment by timoschilling** [comment]
@deivid-rodriguez :+1: can you continue the work on that?

**Comment by deivid-rodriguez** [comment]
I'm on it, I'll open a PR superseding this one. Is that ok?

**Comment by brennovich** [comment]
@deivid-rodriguez even the memory leak specs are passing?

Can you keep with this PR? I think the history here may be important in the future ;)

**Comment by deivid-rodriguez** [comment]
Everything is passing. But I haven't tried `jruby-head`, only `jruby.9.0.0.0.rc2`. I'll let travis do that.

You mean opening a PR at kennethkalmer/activeadmin?

**Comment by deivid-rodriguez** [comment]
@brennovich Note that I am keeping history, I'm just doing it in a new branch in my own fork.

**Comment by brennovich** [comment]
Awesome!

Humm not sure, I think is better to open up a new one, just make sure to comment the new PR number here in the comments :)

Thanks o/

**Comment by deivid-rodriguez** [comment]
Opening a sub-PR is good because commit authors are kept, but just rebasing my work on top of a copy of @kennethkalmer's branch seems easier (and already done).

**Comment by brennovich** [comment]
:+1:

**Comment by deivid-rodriguez** [comment]
Done! And the commit authors are preserved... I love git. :smile: 

Let's see how the build goes.

**Comment by timoschilling** [comment]
closed #4032

**Pull request closed without merging**